src/main/java/org/springframework/retry/annotation/AnnotationAwareRetryOperationsInterceptor.java

```diff
@@ -451,7 +452,8 @@ private BackOffPolicy getBackoffPolicy(Backoff backoff, boolean stateless) {
 		boolean isRandom = false;
 		String randomExpression = (String) attrs.get("randomExpression");
 		Expression parsedRandomExp = null;
-		if (multiplier > 0) {
+
+		if (multiplier > 0 || multiplierExpression != null) {
```

Comment:
Looks like it supposed to be a `parsedMultExp` - exactly the one which is intended for runtime evaluation.
If that is not template, then it is evaluated right now and then `multiplier` is set.